src/inmanta/loader.py

```diff
@@ -662,3 +690,50 @@ def module_in_prefix(module: types.ModuleType, prefix: str) -> bool:
     for mod_name in loaded_modules:
         del sys.modules[mod_name]
     importlib.invalidate_caches()
+
+
+@dataclass(frozen=True)
+class PythonModule:
+    name: str
+    version: str
+    files_in_module: list["SourceInfo"]
+
+
+class SourceInfo(BaseModel):
+    """This class is used to store information related to source code information"""
+
+    # absolute path to this file
+    path: str
+    # python module name e.g. "inmanta_plugins.my_mod.my_submod"
+    module_name: str
+
+    @computed_field  # type: ignore[prop-decorator]
+    @cached_property
+    def hash(self) -> str:
+        """Get the sha1 hash of the file"""
+        sha1sum = hashlib.new("sha1")
+        sha1sum.update(self.content)
+        return sha1sum.hexdigest()
+
+    @cached_property
```

Comment:
1. I think it is important to verify this doesn't leak memory like the `@lrucache`
2. this lazy loading it very tricky (i.e. it works if you are on the same machine, which for testing is always true, but not in general where the compiler can be remote). Can you add a specific test to make sure the file is transported correctly (i.e. not via disk)